AK/IPv4Address.h

```diff
@@ -76,12 +76,37 @@ class [[gnu::packed]] IPv4Address
         if (string.is_null())
             return {};
         auto parts = string.split_view('.');
-        if (parts.size() != 4)
+
+        u32 a;
+        u32 b;
+        u32 c;
+        u32 d;
+
+        if (parts.size() == 1) {
+            a = 0;
+            b = 0;
+            c = 0;
+            d = parts[1].to_uint().value_or(256);
+        } else if (parts.size() == 2) {
+            a = parts[1].to_uint().value_or(256);
+            b = 0;
+            c = 0;
+            d = parts[2].to_uint().value_or(256);
+        } else if (parts.size() == 3) {
+            a = parts[0].to_uint().value_or(256);
+            b = parts[1].to_uint().value_or(256);
+            c = 0;
+            d = parts[2].to_uint().value_or(256);
+        } else if (parts.size() == 4) {
+            a = parts[0].to_uint().value_or(256);
+            b = parts[1].to_uint().value_or(256);
+            c = parts[2].to_uint().value_or(256);
+            d = parts[3].to_uint().value_or(256);
+        } else {
             return {};
-        auto a = parts[0].to_uint().value_or(256);
-        auto b = parts[1].to_uint().value_or(256);
-        auto c = parts[2].to_uint().value_or(256);
-        auto d = parts[3].to_uint().value_or(256);
+        }
+
+
         if (a > 255 || b > 255 || c > 255 || d > 255)
             return {};
         return IPv4Address((u8)a, (u8)b, (u8)c, (u8)d);
```

Comment:
I know you don't intend to support obscure forms like `1.65793`. But i would suggest that you should try to do it in this PR. It's actually quite a simple change to `IPv4Address::IPv4Address` with some bitwise arithmetic.

See the spoiler below if you want to know how I would do it. If not, thats fine too.

<details> 
  <summary> SPOILER</summary>
   <pre>
IPv4Address(u8 a, u8 b, u8 c, u32 d)
{
    m_data[0] = a | ((d >> 24) & 0xff);
    ...
}
   </pre>
</details>

Reply:
I'm not familiar with this operating system and writing networking-related code yet. If I were comfortable with doing so, I'd have done it in the first place. I'd much rather do this at a later stage, which is what I'm currently preparing for, after having discussed the details with other people, rather than trying to act based on a code sample that I'm not sure I even understand.